src/components/Drawer.jsx

```diff
@@ -13,86 +13,82 @@ const sizes = {
     "2xl": "w-screen md:max-w-screen-md 2xl:max-w-screen-lg", // This was the old size
 };
 
-export const Drawer = ({
-    isOpen = false,
-    title,
-    size = "medium",
-    content,
-    onClose,
-    classNames = {},
-    position = "right",
-}) => {
-    return (
-        <Transition.Root show={isOpen} as={Fragment}>
-            <Dialog
-                as="div"
-                className={clsx("ui-drawer fixed inset-0 z-10 overflow-hidden", classNames.dialog)}
-                open={isOpen}
-                onClose={onClose}
-            >
-                <div className="flex h-screen w-full">
-                    <Transition.Child
-                        as={Fragment}
-                        enter="ease-in-out duration-500"
-                        enterFrom="opacity-0"
-                        enterTo="opacity-100"
-                        leave="ease-in-out duration-500"
-                        leaveFrom="opacity-100"
-                        leaveTo="opacity-0"
-                    >
-                        <Dialog.Overlay className="absolute inset-0 bg-gray bg-opacity-75 transition-opacity" />
-                    </Transition.Child>
-                    <div
-                        className={clsx(
-                            "fixed inset-y-0 flex max-w-full",
-                            position === "right" ? "right-0" : "left-0",
-                            classNames.dialogContent,
-                        )}
-                    >
+export const Drawer = forwardRef(
```

Comment:
why is forwardRef needed ?

Reply:
Rushi asked to add it

Reply:
@manojx031 We are adding forwardRef on all of our primary components so that we can pass in refs from seller app to modify things. It's additional power for us. I've seen all of the UI libraries doing this.